frontend/src/utils/keycloakHook.js

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useRef } from 'react';
+import { useKeycloak } from '@react-keycloak/web';
+
+const useCustomKeycloak = ({ customNonce } = {}) => {
+  let nonce = customNonce;
+  if (!nonce) {
+    customNonce = createUUID();
+  }
+  const { keycloak, initialized } = useKeycloak();
+  const origCreateLoginUrl = keycloak.createLoginUrl;
+  keycloak.createLoginUrl = (options) => {
+    let url = origCreateLoginUrl(options);
+    if (!url.includes('&nonce=')) {
+      url = url + '&nonce=' + encodeURIComponent(nonce);
+    }
+    return url;
+  };
+  return { keycloak, initialized };
+};
+
+//----------------------------------------------------------------------------------
+//functions in this section are from keycloak-js
+
+function createUUID() {
```

Comment:
Right I see it's not possible to import this function with scoping in Keycloak. It might be easier if we just add this lib: https://www.npmjs.com/package/uuid

Reply:
It seems it's creating a v4 uuid if it's 36 characters `uuidv4`

Reply:
will do!